src/linear_solver_logs/solve_log_full_ma.jl

```diff
@@ -0,0 +1,352 @@
+# This file is part of RLinearAlgebra.jl
+
+# using LinearAlgebra
+"""
+    MAInfo
+
+A mutable structure that stores information relevant to the moving average of the progress estimator.
+
+# Fields
+- `lambda1::Int64`, the width of the moving average during the fast convergence phase of the algorithm,
+  this has a default value of one.    
+- `lambda2::Int64`, the width of the moving average in the slower convergence phase, this has a default value of 30.
+- `lambda::Int64`, the value of the moving average width at the current iteration.
+- `flag::Bool`, A boolean indicating which phase we are in, a value of `true` indicates second phase. 
+- `idx::Int64`, The index indcating what value should be replaced in the moving average buffer.
+- `res_window::Vector{Float64}` - The moving average buffer.
+"""
+mutable struct MAInfo
+    lambda1::Int64
+    lambda2::Int64
+    lambda::Int64
+    flag::Bool
+    idx::Int64
+    res_window::Vector{Float64}
+end
+
+""" 
+    DistInfo
+
+A mutable structure that stores information about the sub-Exponential distribution.
+
+# Fields
+- `sampler::Union{DataType, Nothing}`, The type of sampler being used.
+- `max_dimension::Int64`, The dimension that is being sampled.
+- `block_dimension::Int64`, The sampling dimension.
+- `sigma2::Union{Float64, Nothing}`, The variance parameter in the sub-Exponential distribution, 
+   if not given is determined for sampling method.
+- `omega::Union{Float64, Nothing}`, The exponential distrbution parameter, if not given is determined for sampling methods.
+- `eta::Float64`, A parameter for adjusting the conservativeness of the distribution, higher value means a less conservative
+  estimate. By default, this is set to one.
+"""
+mutable struct DistInfo
+    sampler::Union{DataType, Nothing}
+    max_dimension::Int64
+    block_dimension::Int64
+    sigma2::Union{Float64, Nothing}
```

Comment:
so above you have `lambda2` and here you have `sigma2`. In one case you mean $\lambda_2$ and the other case you mean $\sigma^2$. 

Is this something that we should think about when it comes to consistency? I think it can work as is, but wanted to put the question out. @dmaldona I do not think we should change it here, it is a v0.2 question.

Reply:
Ok good thing to keep in mind.